File: backend/java/localpos-product-service/src/main/java/com/pds/localpos/productservice/config/ExceptionHandlerConfig.java
package com.pds.localpos.productservice.config;

import com.pds.localpos.common.handler.GlobalExceptionHandler;
import com.pds.localpos.common.i18n.ErrorMessageResolver;
import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.util.Locale;

@Configuration
@RequiredArgsConstructor
public class ExceptionHandlerConfig {

    private final ErrorMessageResolver errorMessageResolver;
    private final I18nConfigurationProperties i18nConfigurationProperties;

    @Bean
    public GlobalExceptionHandler globalExceptionHandler() {
        Locale locale = i18nConfigurationProperties.getLocale();
        return new GlobalExceptionHandler(errorMessageResolver, locale);
    }
}

File: backend/kotlin/localpos-user-service-kt/src/main/kotlin/com/pds/localpos/userservice/config/ExceptionHandlerConfig.kt
package com.pds.localpos.userservice.config

import com.pds.localpos.common.handler.GlobalExceptionHandler
import com.pds.localpos.common.i18n.ErrorMessageResolver
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class ExceptionHandlerConfig(
    private val errorMessageResolver: ErrorMessageResolver,
    private val i18nConfigurationProperties: I18nConfigurationProperties
) {

    @Bean
    fun globalExceptionHandler(): GlobalExceptionHandler {
        return GlobalExceptionHandler(
            errorMessageResolver,
            i18nConfigurationProperties.resolvedLocale
        )
    }
}

